syntax = "proto3";

package WorkerArgs;

message CreateAgentArgs {
    string agent_id = 1;
    bytes agent_init_args = 2;
    bytes agent_source_code = 3;
};

message AgentArgs {
    string agent_id = 1;
};

// message ReplyArgs {
//     string agent_id = 1;
//     bytes message = 2;
//     int32 task_id = 3;
//     int32 callback_id = 4;
// };

// message ObserveArgs {
//     string agent_id = 1;
//     bytes message = 2;
// }

message AgentFuncArgs {
    string agent_id = 1;
    string func_name = 2;
    bytes raw_value = 3;
}

message ModelConfigsArgs {
    bytes model_configs = 1;
}

message AgentListReturn {
    repeated string agent_str_list = 1;
}

message MsgReturn {
    bool ok = 1;
    string message = 2;
}
